addons/account/models/account_move.py

```diff
@@ -2946,31 +3010,58 @@ def create(self, vals_list):
         BUSINESS_FIELDS = ('price_unit', 'quantity', 'discount', 'tax_ids')
+        PROTECTED_FIELDS_TAX_LOCK_DATE = ['debit', 'credit', 'tax_line_id', 'tax_ids', 'tag_ids']
+        PROTECTED_FIELDS_LOCK_DATE = PROTECTED_FIELDS_TAX_LOCK_DATE + ['account_id', 'journal_id', 'amount_currency', 'currency_id', 'partner_id']
+        PROTECTED_FIELDS_RECONCILIATION = ('account_id', 'date', 'debit', 'credit', 'amount_currency', 'currency_id')
 
         if ('account_id' in vals) and self.env['account.account'].browse(vals['account_id']).deprecated:
             raise UserError(_('You cannot use a deprecated account.'))
-        if any(key in vals for key in ('account_id', 'journal_id', 'date', 'move_id', 'debit', 'credit')):
-            self._update_check()
-        if not self._context.get('allow_amount_currency') and any(
-                key in vals for key in ('amount_currency', 'currency_id')):
-            # hackish workaround to write the amount_currency when assigning a payment to an invoice through the 'add' button
-            # this is needed to compute the correct amount_residual_currency and potentially create an exchange difference entry
-            self._update_check()
         # when making a reconciliation on an existing liquidity journal item, mark the payment as reconciled
-        for record in self:
-            if 'statement_line_id' in vals and record.payment_id:
+        for line in self:
+            if line.parent_state == 'posted':
+                if line.move_id.restrict_mode_hash_table and set(vals).intersection(INTEGRITY_HASH_LINE_FIELDS):
+                    raise UserError(_("You cannot edit the following fields due to restrict mode being activated on the journal: %s.") % ', '.join(INTEGRITY_HASH_LINE_FIELDS))
+                if any(key in vals for key in ('tax_ids', 'tax_line_ids')):
+                    raise UserError(_('You cannot modify the taxes related to a posted journal item, you should reset the journal entry to draft to do so.'))
+            if 'statement_line_id' in vals and line.payment_id:
                 # In case of an internal transfer, there are 2 liquidity move lines to match with a bank statement
-                if all(line.statement_id for line in record.payment_id.move_line_ids.filtered(
-                        lambda r: r.id != record.id and r.account_id.internal_type == 'liquidity')):
-                    record.payment_id.state = 'reconciled'
+                if all(line.statement_id for line in line.payment_id.move_line_ids.filtered(
+                        lambda r: r.id != line.id and r.account_id.internal_type == 'liquidity')):
+                    line.payment_id.state = 'reconciled'
+
+            # Check move consistency.
+            check_lock_date = any(field_will_change(line, field_name) for field_name in PROTECTED_FIELDS_LOCK_DATE)
+            line.move_id._check_move_consistency(check_balanced=False, check_lock_date=check_lock_date, check_tax_lock_date=False)
```

Comment:
directly call check_lock_date()

Reply:
No! I want to check the tax lock date for this line only. Example: I'm writing an additional line not affecting the taxes => it should work.